src/DynamicData.Tests/List/MergeManyCacheChangeSetsFixture.cs

```diff
@@ -0,0 +1,737 @@
+        results.Data.Count.Should().Be(PricesPerMarket);
+        results.Messages.Count.Should().Be(1);
+        results.Summary.Overall.Adds.Should().Be(PricesPerMarket);
+        results.Summary.Overall.Removes.Should().Be(0);
+        results.Summary.Overall.Updates.Should().Be(0);
+        results.Summary.Overall.Refreshes.Should().Be(0);
+    }
+
+    [Fact]
+    public void EveryItemVisibleWhenSequenceCompletes()
+    {
+        // having
+        _marketList.AddRange(Enumerable.Range(0, MarketCount).Select(n => new FixedMarket(GetRandomPrice, n * ItemIdStride, (n * ItemIdStride) + PricesPerMarket)));
+
+        // when
+        using var results = _marketList.Connect().MergeManyChangeSets(m => m.LatestPrices).AsAggregator();
+        DisposeMarkets();
+
+        // then
+        results.Data.Count.Should().Be(PricesPerMarket * MarketCount);
+        results.Summary.Overall.Adds.Should().Be(PricesPerMarket * MarketCount);
+        results.Summary.Overall.Removes.Should().Be(0);
+        results.Summary.Overall.Updates.Should().Be(0);
+        results.Summary.Overall.Refreshes.Should().Be(0);
+    }
+
+    [Theory]
+    [InlineData(false, false)]
+    [InlineData(false, true)]
+    [InlineData(true, false)]
+    [InlineData(true, true)]
+    public void MergedObservableCompletesOnlyWhenSourceAndAllChildrenComplete(bool completeSource, bool completeChildren)
+    {
+        // having
+        _marketList.AddRange(Enumerable.Range(0, MarketCount).Select(n => new FixedMarket(GetRandomPrice, n * ItemIdStride, (n * ItemIdStride) + PricesPerMarket, completable: completeChildren)));
+        var hasSourceSequenceCompleted = false;
+        var hasMergedSequenceCompleted = false;
+
+        using var cleanup = _marketList.Connect().Do(_ => { }, () => hasSourceSequenceCompleted = true)
+                            .MergeManyChangeSets(m => m.LatestPrices).Subscribe(_ => { }, () => hasMergedSequenceCompleted = true);
+
+        // when
+        if (completeSource)
+        {
+            DisposeMarkets();
+        }
+
+        // then
+        hasSourceSequenceCompleted.Should().Be(completeSource);
+        hasMergedSequenceCompleted.Should().Be(completeSource && completeChildren);
+    }
+
+    [Fact]
+    public void MergedObservableWillFailIfSourceFails()
+    {
+        // having
+        var markets = Enumerable.Range(0, MarketCount).Select(n => new Market(n)).ToArray();
+        _marketList.AddRange(markets);
+        var receivedError = default(Exception);
+        var expectedError = new Exception("Test exception");
+        var throwObservable = Observable.Throw<IChangeSet<IMarket>>(expectedError);
+
+        using var cleanup = _marketList.Connect().Concat(throwObservable)
+                            .MergeManyChangeSets(m => m.LatestPrices).Subscribe(_ => { }, err => receivedError = err);
+
+        // when
+        DisposeMarkets();
+
+        // then
+        receivedError.Should().Be(expectedError);
+    }
+
```

Comment:
Following off of the bug I fixed yesterday, could/should we add a test to cover downstream-teardown (I.E. unsubscription) behavior? Internally, this means cleaning up all the outstanding child stream subscriptions, right?

Reply:
We probably should do.  I wonder whether we should raise an issue and gradually trawl through the system.

Reply:
I wouldn't expect that there's a systemic issue with not handling downstream-teardown, that was just a defect I introduced a few days ago. But it is a behavior worth testing, wherever we have resources that need cleanup.

Perhaps instead what we need is a guidelines document for testing operators.